### PR TITLE
Docs refactor to cgroups style

### PR DESCRIPTION
**Pull request by Phil-Friderici**
Changed the layout to match the style from the cgroups module.
- add TOC
- add datatype where missing
- add valid values when validate_re was used

Missing:
- parameter description for $bind::forwarders
- short description for class bind
- short description for define bind::view
- short description for define bind::zone

### CODE REVIEW

**Comment by Phil-Friderici** [comment]
Sorry for mixing this PR with updates to the testing rig. If needed, I can separate then.

File: README.md [hunk 1]
```diff
@@ -434,226 +455,226 @@ bind::controls:
...
 
-===
+## Defined type `bind::masters`
 
-# Parameters for `define bind::masters`
+### Description
 
 Manage a masters declaration.
 
-name (type: String)
```
Comment (ghoneycutt) [inline README.md]: why is this removed? it is elsewhere or no longer needed due to code changes?

File: README.md [hunk 2]
```diff
@@ -434,226 +455,226 @@ bind::controls:
...
-path (type: Integer)
 ---
+#### path (type: Integer)
 Absolute path to file containing the key declaration.
 
-- *Default*: "/etc/named/{name}.key"
-
-```
-# example
-bind::keys:
```
Comment (ghoneycutt) [inline README.md]: why is this removed? it is elsewhere or no longer needed due to code changes?

File: Gemfile [hunk 3]
```diff
@@ -7,33 +7,23 @@ else
 end
 
 gem 'metadata-json-lint'
-gem 'puppetlabs_spec_helper', '>= 1.1.1'
-gem 'puppet-lint', '>= 1.0.0'
+gem 'puppetlabs_spec_helper', '>= 1.2.0'
 gem 'facter', '>= 1.7.0'
-gem 'rspec-puppet', '~> 2.0'
+gem 'rspec-puppet'
+gem 'puppet-lint', '>= 1.0', '< 3.0'
```
Comment (ghoneycutt) [inline Gemfile]: all of the other puppet-lint gems are v2 compat now, so this can be

```ruby
gem 'puppet-lint', '~> 2.0'
```

File: spec/defines/key_spec.rb [hunk 4]
```diff
@@ -129,20 +131,10 @@
...
-        #:fact => 'value',
-      }
-    end
-    let(:mandatory_params) do
-      {
-        :secret => 'geheim',
-      }
-    end
+    let(:facts) { mandatory_facts }
+    let(:mandatory_params) { { :secret => 'geheim' } }
```
Comment (ghoneycutt) [inline spec/defines/key_spec.rb]: does this syntax do a merge of the mandatory_params hash with the hash given?

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -11,3 +11,13 @@
     Facter.clear_messages
   end
 end
+
+def mandatory_facts
```
Comment (ghoneycutt) [inline spec/spec_helper.rb]: great idea!

File: README.md (hunk 1 above)
Comment (Phil-Friderici) [inline README.md]: This moved to line 693 as I moved the examples to come before the parameter descriptions.

File: README.md (hunk 2 above)
Comment (Phil-Friderici) [inline README.md]: Moved the examples on top of the parameter descriptions. This moved to line 638.

File: Gemfile (hunk 3 above)
Comment (Phil-Friderici) [inline Gemfile]: done

File: spec/defines/key_spec.rb (hunk 4 above)
Comment (Phil-Friderici) [inline spec/defines/key_spec.rb]: no, it creates a new set of :mandatory_params here
